helpers/utils.py

```diff
@@ -107,4 +128,117 @@ def UmaskNamedTemporaryFile(*args, **kargs):
     umask = os.umask(0o666)
     os.umask(umask)
     os.chmod(fdesc.name, 0o666 & ~umask)
-    return fdesc
\ No newline at end of file
+    return fdesc
+
+def build_kubectl_cmd(namespace: str, k8s_context: str) -> str:
+    kubeconfig = get_kube_config()
+    kubectl_cmd = get_kubectl_cmd()
+    cmd = f"{kubectl_cmd} --kubeconfig {kubeconfig} --namespace={namespace}"
+    if k8s_context:
+        cmd += f" --context {k8s_context}"
+    return cmd
+
+def build_secret_name(username: str, namespace: str) -> str:
```

Comment:
you don't need namespace, do you?

username should be Optional (inferring this from your code "username or spark")